### PR TITLE
ci-operator: optionally use pod-utils to push artifacts up

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @alvaroaleman

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
testing this with `skip_report` here: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/497/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1350155918758645760

**Comment by alvaroaleman** [comment]
/unassign
ping me when its ready, it creates too many notifications right now

**Comment by stevekuznetsov** [comment]
Latest output here: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/497/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-test/1351287537708044288

The artifacts layout is now pretty much the same as it was before, and everything looks good.

**Comment by stevekuznetsov** [comment]
/retitle ci-operator: optionally use pod-utils to push artifacts up

**Comment by stevekuznetsov** [comment]
/hold

I need to:

 - [ ] update `openshift/release` to use `$ARTIFACT_DIR` instead of `/tmp/artifacts` literals
 - [ ] update `ci-operator` release `Pod` config to do the same

**Comment by bbguimaraes** [comment]
@stevekuznetsov: I'd also check how that impacts test pod scheduling since we'll now do the "copy binary, add init container" dance twice (`secret-wrapper` + `entrypoint`) and a single one can sometimes take forever in our clusters.

Are images for init containers pulled in parallel or serially once each is scheduled?

File: pkg/steps/artifacts.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ import (
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/remotecommand"
+	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
```
Comment (alvaroaleman) [inline pkg/steps/artifacts.go:28]: please avoid import names like `v1`, they make the code confusing to read :upside_down_face:

File: pkg/steps/artifacts.go [hunk 2]
```diff
@@ -340,6 +342,30 @@ func removeFile(podClient PodClient, ns, name, containerName string, paths []str
 	return nil
 }
 
+func addPodUtils(pod *coreapi.Pod, artifactDir string, decorationConfig *v1.DecorationConfig, rawJobSpec string) error {
```
Comment (alvaroaleman) [inline pkg/steps/artifacts.go:345]: Could you add/extend a fixture-based test for this somewhere? Seeing this as yaml makes reasoning much easier than just the code

File: pkg/steps/artifacts.go (hunk 1 above)
Comment (stevekuznetsov) [reply pkg/steps/artifacts.go:28]: I blame GoLand ;)

**Comment by alvaroaleman** [comment]
> @stevekuznetsov: I'd also check how that impacts test pod scheduling since we'll now do the "copy binary, add init container" dance twice (secret-wrapper + entrypoint) and a single one can sometimes take forever in our clusters.

We should probably start creating BZs around that. This is an openshift-specific problem (or heavy-loaded openshift specific? Not sure)

**Comment by stevekuznetsov** [comment]
Part of DPTP-1819

File: pkg/steps/artifacts.go (hunk 2 above)
Comment (stevekuznetsov) [reply pkg/steps/artifacts.go:345]: done

**Comment by stevekuznetsov** [comment]
@alvaroaleman ready for a look now

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**